<launch>
  <arg name="input_topic_name" default="/velodyne_points_ex"/>
  <arg name="output_topic_name" default="/pointcloud_filtered"/>

  <arg name="input_frame" default="velodyne"/>
  <arg name="output_frame" default="velodyne"/>
  <arg name="visibility_threshold" default="0.95"/>
  <arg name="vertical_bins" default="128"/>
  <arg name="max_azimuth_diff" default="50.0"/>

  <node pkg="pointcloud_preprocessor" exec="dual_return_outlier_filter_node" name="dual_return_outlier_filter">
    <remap from="input" to="$(var input_topic_name)"/>
    <remap from="output" to="$(var output_topic_name)"/>

    <param name="input_frame" value="$(var input_frame)"/>
    <param name="output_frame" value="$(var output_frame)"/>
    <param name="visibility_threshold" value="$(var visibility_threshold)"/>
    <param name="vertical_bins" value="$(var vertical_bins)"/>
    <param name="max_azimuth_diff" value="$(var max_azimuth_diff)"/>
  </node>
</launch>
